source/common/upstream/health_checker_impl.h

```diff
@@ -226,6 +226,14 @@ class HttpHealthCheckerImpl : public HealthCheckerImplBase {
 
   typedef std::unique_ptr<HttpActiveHealthCheckSession> HttpActiveHealthCheckSessionPtr;
 
+  struct HeaderValueWrapper {
```

Comment:
Can we share this code in a utility class somehow? I think this was mostly copied from the router or elsewhere? (Same as the config/setting functions above). I suspect we will be here again somewhere else in the future...

Reply:
> I think this was mostly copied from the router or elsewhere?

Reply:
@mattklein123 I moved it into `common/http/utility` if it is not the appropriate place, please let me know (probably just go ahead to add `HealthCheckUtility` class?). Thanks!